js/main.js

```diff
@@ -0,0 +1,40 @@
+var values = [[1, "rt"], [2, true], [3, false], [4, 1], [5, {}], [6, "2"], [7], []];
+
+autoCalc();
+
+function autoCalc() {
+  for (i = 0; i < values.length; i++) {
+    var x = values[i][0];
+    var y = values[i][1];
+    logAction(x, y, add);
+    logAction(x, y, sub);
+    logAction(x, y, multiply);
+    logAction(x, y, divide);
+    logAction(x, y, divideInteger);
+    console.log('\n');
+  }
+}
+
+function logAction(x, y, func) {
```

Comment:
I hope you remember and understand mechanism of hoisting and reason you can use  this func in `autoCalc`